galaxy/tools/deps/mulled/mulled_update_singularity_containers.py

```diff
@@ -0,0 +1,321 @@
+    # >>> get_test('snakemake:3.11.2--py34_1')
+    # {'commands': ['snakemake --help > /dev/null'], 'imports': ['snakemake'], 'import_lang': 'python -c', 'container': 'snakemake:3.11.2--py34_1'}
+    # >>> get_test('perl-yaml:1.15--pl5.22.0_0')
+    # {'imports': ['YAML', 'YAML::Any', 'YAML::Dumper', 'YAML::Dumper::Base', 'YAML::Error', 'YAML::Loader', 'YAML::Loader::Base', 'YAML::Marshall', 'YAML::Node', 'YAML::Tag', 'YAML::Types'], 'import_lang': 'perl -e', 'container': 'perl-yaml:1.15--pl5.22.0_0'}
+    """
+
+    package_tests = {}
+    name = container.replace('--', ':').split(':') # list consisting of [name, version, (build, if present)]
+
+    r = requests.get("https://anaconda.org/bioconda/%s/%s/download/linux-64/%s.tar.bz2" % (name[0], name[1], '-'.join(name)))
+    
+    try:
+        tarball = tarfile.open(mode="r:bz2", fileobj=BytesIO(r.content))
+    except tarfile.ReadError:
+        pass
+    else:
+        try: # try to open meta.yaml
+            metafile = tarball.extractfile('info/recipe/meta.yaml')
+            
+        except KeyError: # if it's not there ...
+            logging.error("meta.yaml file not present.")
+        else:
+            meta_yaml = yaml.load(Template(metafile.read().decode('utf-8')).render()) # run the file through the jinja processing
+            try:
+                if meta_yaml['test']['commands'] != [None]:
+                    package_tests['commands'] = meta_yaml['test']['commands']
+            except KeyError:
+                pass
+
+            try:
+                if meta_yaml['test']['imports'] != [None]:
+                    package_tests['imports'] = meta_yaml['test']['imports']
+            except KeyError:
+                pass
+            
+            #need to know what scripting languages are needed to run the container
+            try:
+                requirements = list(meta_yaml['requirements']['run'])
+            except (KeyError, TypeError):
+                pass
+            else:
+                for requirement in requirements:
+                    if requirement.split()[0] == 'perl':
+                        package_tests['import_lang'] = 'perl -e'
+                        break
+                    # elif ... :
+                        # other languages if necessary ... hopefully python and perl should suffice though
+                else: # python by default
+                    package_tests['import_lang'] = 'python -c'
+
+        if not package_tests: # if meta.yaml was not present or there were no tests in it, try and get run_test.sh instead
+            try:
+                run_test = tarball.extractfile('info/recipe/run_test.sh')
+                package_tests['commands'] = run_test.read()
+            except KeyError:
+                logging.error("run_test.sh file not present.")
+
+    package_tests['container'] = container
+    return package_tests # {'commands': ...}
+
+
+def mulled_get_test(container):
+    """
+    Gets test for hashed containers
```

Comment:
Mh, I think you want to find the original name/packages underlying the mulled containers right? in a second step you want to impot the get_tests function from above and get tests for those.